include/shell/shell.h

```diff
@@ -36,6 +36,21 @@ extern "C" {
 
 #define SHELL_CMD_ROOT_LVL		(0u)
 
+/** @brief Flag indicates that optional arguments will be treated as one,
+ *	   unformatted argument.
+ *
+ * By default, shell is parsing all arguments, treats all spaces as argument
+ * separators unless they are within quotation marks which are removed in that
+ * case. If command rely on unformatted argument then this flag shall be used
+ * in place of number of optional arguments in command definition to indicate
+ * that only mandatory arguments shall be parsed and remaining command string is
+ * passed as a raw string.
+ */
+#define SHELL_OPT_ARG_RAW	(0xFE)
+
+/** @brief Flag indicating maximum number of arguments. */
+#define SHELL_ARG_MAX		(0xFD)
```

Comment:
This name is quite similar to `SHELL_ARGC_MAX` in Kconfig.

Reply:
renamed to `SHELL_OPT_ARG_MAX`.